misc.go

```diff
@@ -42,6 +43,14 @@ func (s WebError) Error() string {
 	return string(s)
 }
 
+type RateLimitedError struct {
```

Comment:
lets add a method Backoff() time.Duration and change RetryAfter to time.Duration.

then people can do the following:
```
type backoff interface {
  Backoff() time.Duration
}

switch cause := err.(type) {
case backoff:
  time.Sleep(cause.Backoff())
}
```

Reply:
[`backoff.go` already has a `backoff` struct](https://github.com/nlopes/slack/blob/master/backoff.go#L15), should we consider a different name?

Reply:
I for one like the "RetryAfter" name. It's very hard to misunderstand and is consistent with the header name.

Reply:
@james-lawrence is it ok by you if we go ahead and merge this as-is, i.e. can we PR the `Backoff()`/`RetryAfter()` in a separate PR?

Reply:
absolutely